### PR TITLE
SSL certs

### PR DESCRIPTION
**Pull request by IncredibleRichie**
Added SSL handling to products and repositories

### CODE REVIEW

File: plugins/modules/katello_repository.py [hunk 1]
```diff
@@ -117,6 +132,11 @@
       - comma separated list of architectures to be synced from deb-archive
       - only available for I(content_type=deb)
     type: str
+  deb_errata_url:
+    description:
+      - Errataurl to sync from
+    type: string
```
Comment (evgeni) [inline plugins/modules/katello_repository.py]:
```suggestion
    type: str
```

File: plugins/modules/katello_repository.py [hunk 2]
```diff
@@ -69,6 +69,21 @@
     - Repository GPG key
     required: false
     type: str
+  ssl_ca_cert:
+    description:
+    - Repository SSL CA CERT
```
Comment (evgeni) [inline plugins/modules/katello_repository.py]:
```suggestion
    - Repository SSL CA certificate
```

File: plugins/modules/katello_repository.py [hunk 3]
```diff
@@ -69,6 +69,21 @@
...
+    required: false
+    type: str
+  ssl_client_cert:
+    description:
+    - Repository SSL CLIENT CERT
```
Comment (evgeni) [inline plugins/modules/katello_repository.py]:
```suggestion
    - Repository SSL client certificate
```

File: plugins/modules/katello_repository.py [hunk 4]
```diff
@@ -69,6 +69,21 @@
...
+    required: false
+    type: str
+  ssl_client_key:
+    description:
+    - Repository SSL CLIENT KEY
```
Comment (evgeni) [inline plugins/modules/katello_repository.py]:
```suggestion
    - Repository SSL client private key
```

File: plugins/modules/katello_repository.py [hunk 5]
```diff
@@ -117,6 +132,11 @@
       - comma separated list of architectures to be synced from deb-archive
       - only available for I(content_type=deb)
     type: str
+  deb_errata_url:
+    description:
+      - Errataurl to sync from
```
Comment (evgeni) [inline plugins/modules/katello_repository.py]:
```suggestion
      - URL to sync errata information from
```

File: plugins/modules/katello_product.py [hunk 6]
```diff
@@ -50,6 +50,21 @@
     - Content GPG key name attached to this product
     required: false
     type: str
+  ssl_ca_cert:
+    description:
+    - Content SSL CA CERT name attached to this product
```
Comment (evgeni) [inline plugins/modules/katello_product.py]:
```suggestion
    - Content SSL CA certificate name attached to this product
```

File: plugins/modules/katello_product.py [hunk 7]
```diff
@@ -50,6 +50,21 @@
...
+    required: false
+    type: str
+  ssl_client_cert:
+    description:
+    - Content SSL CLIENT CERT name attached to this product
```
Comment (evgeni) [inline plugins/modules/katello_product.py]:
```suggestion
    - Content SSL client certificate name attached to this product
```

File: plugins/modules/katello_product.py [hunk 8]
```diff
@@ -50,6 +50,21 @@
...
+    required: false
+    type: str
+  ssl_client_key:
+    description:
+    - Content SSL CLIENT KEY name attached to this product
```
Comment (evgeni) [inline plugins/modules/katello_product.py]:
```suggestion
    - Content SSL client private key name attached to this product
```

**Comment by IncredibleRichie** [comment]
Thank you very much. I'll take your names.
Regarding the tests, unfortunately I have not yet accumulated enough knowledge to write them

**Comment by evgeni** [comment]
> Regarding the tests, unfortunately I have not yet accumulated enough knowledge to write them

I heard rumors, there are people in your office, who have done so ;-)
And I can mentor you remotely too.
The question is: do you want to? :)

**Comment by IncredibleRichie** [comment]
^^
It would like to learn it. 
If you don't mind not doing it today. (unfortunately no time left)

**Comment by evgeni** [comment]
No worries, this does not look like an urgent change.

**Comment by mdellweg** [comment]
I'll take over test guidance here, due to geographical reasons. ;)

**Comment by evgeni** [comment]
> I'll take over test guidance here, due to geographical reasons. ;)

Thanks!

File: tests/test_playbooks/repository.yml [hunk 9]
```diff
@@ -67,6 +67,9 @@
           repository_name: "Test Repository 2"
           repository_label: "just_a_test_repo2"
           repository_gpg_key: "Test GPG key"
+          repository_ssl_ca_cert: "Test GPG key"
```
Comment (evgeni) [inline tests/test_playbooks/repository.yml]: Well, I call this cheating, but clever ;)

**Comment by evgeni** [comment]
Can you please rebase this now that #610 is merged? Other than the rebase this is good to go!

**Comment by evgeni** [comment]
(for the rebase, you can drop the `flat_name` definitions, those are automatic now, and you don't have to call `exit_json` anymore)

**Comment by mdellweg** [comment]
I think @evgeni s requested changes have been addressed. Test were added.

**Comment by evgeni** [comment]
Thanks @IncredibleRichie!

**Pull request merged**